docs/source/topics/data_types.rst

```diff
@@ -305,14 +305,47 @@ This class can be initialized via the **absolute** path to the file you want to
 
   In [2]: single_file = SinglefileData('/absolute/path/to/file')
 
+When storing the node, the filename is stored in the database and the file itself is copied to the repository.
 The contents of the file in string format can be obtained using the :py:meth:`~aiida.orm.nodes.data.singlefile.SinglefileData.get_content()` method:
 
 .. code-block:: ipython
 
   In [3]: single_file.get_content()
   Out[3]: 'The file content'
 
-When storing the node, the filename is stored in the database and the file itself is copied to the repository.
+For large files, reading the entire content into memory using :py:meth:`~aiida.orm.nodes.repository.NodeRepository.get_object_content()` may not be desirable.
+Instead, a file-like handle can be opened to a file in the repository which can be used to read the content as a stream:
+
+.. code-block:: ipython
+
+    In [4]: with single_file.open() as handle:
+            print(handle.read())
+    Out[4]: 'The file content'
```

Comment:
While simple, this is not a good example IMHO, because it ends up anyway putting everything in memory in `handle.read()`, defeating the purpose and making it not clear why we have to go via the new interface.

We should thing at something simple but really streaming. E.g. just pseudo code:
```python
CHUNK_SIZE = 65536
with single_file.open() as handle:
    while True:
        chunk = handle.read(CHUNK_SIZE)
        if not chunk: 
            break
        #process chunk here
```

Or something simple, like counting length:
```python
CHUNK_SIZE = 65536

length = 0
with single_file.open() as handle:
    while True:
        chunk = handle.read(CHUNK_SIZE)
        if not chunk: 
            break
        length += len(chunk)
```

Reply:
OK!